### PR TITLE
!!! TASK: Make kickstart:site command work with Neos 9.0

### PR DESCRIPTION
**Pull request by ahaeslich**
**Review instructions**

Run `./flow kickstart:site My.Site` and create a new site based on this package via `./flow site:create my-site My.Site My.Site:Document.Homepage`.

Tested based on our Neos 9 Beta 10 Release.

Related Issue: #4016

**Upgrade instructions**

Custom packages implementing the `SitePackageGeneratorInterface` will need to be adjusted as we don't provide the `siteName` anymore.

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

**Comment by ahaeslich** [comment]
We briefly talked about the site kickstarter in our last weekly and we might want to do a follow up PR that e.g. includes an import as before. With this PR at least one can create a site package again.

**Review by kitsunet — APPROVED** [review]

**Review by mhsdesign — APPROVED** [review]
Thanks for the adjustments.

I also added php stan level 8 to the site kickstarter codebase so it doesnt rot again ;)

After testing it seems to work, though we dont make use of the `site name` option by default anymore

```
./flow kickstart:site <package key> <site name>
```

File: Neos.SiteKickstarter/Resources/Private/AfxGenerator/NodeTypes/Document.Homepage.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# This is a custom root node type which has been auto-generated for your site package
+# by Neos. You can customize this to your needs.
+'{packageKey}:Document.Homepage':
+  superTypes:
+    'Neos.Neos:Site': true
+  ui:
+    icon: globe
+    label: '{packageKey} Homepage'
```
Comment (kdambekalns) [inline Neos.SiteKickstarter/Resources/Private/AfxGenerator/NodeTypes/Document.Homepage.yaml:8]: This could be the (human-readable) site name, no?

**Review by kdambekalns — COMMENTED** [review]
> After testing it seems to work, though we dont make use of the site name option by default anymore

Why? I guess it was used in the generated XML, that we no longer have. If it's indeed unused (I suggest one possible place to use it (again), though), we should remove the option!

File: Neos.SiteKickstarter/Resources/Private/AfxGenerator/NodeTypes/Document.Homepage.yaml (hunk 1 above)
Comment (ahaeslich) [reply Neos.SiteKickstarter/Resources/Private/AfxGenerator/NodeTypes/Document.Homepage.yaml:8]: I would rather remove the option tbh.

**Review by mhsdesign — CHANGES_REQUESTED** [review]
One more thing: As we require the home page to be `Neos.Neos:Site` see https://github.com/neos/rector/issues/66 we have to auto generate this node type so it can be used to create a site via

```
flow kickstart:site My.Site
flow site:create my-site My.Site My.Site:Document.Homepage
```

Additionally the kickstarter should probably prefer the `NodeTypes` folder for the new homepage nodetype

**Review by mhsdesign — APPROVED** [review]
Lol i didnt use your branch when i checked to request changes. I see everything is perfect then ❤️

**Pull request merged**